### PR TITLE
Mikayla rest apis

### PR DESCRIPTION
**Pull request by me-louie**
- added GET, PUT, and POST endpoints (user-controller)
- created tentative user schema which needs to be validated

### CODE REVIEW

File: server/models/user-model.js [hunk 1]
```diff
@@ -12,3 +12,36 @@ const Schema = mongoose.Schema;
...
+const Setting = new Schema({
+  _id: { type: String, required: true },
+  notification: { type: Boolean, required: true },
+  darkMode: { type: Boolean, required: true },
+  language: { type: String, required: true },
+  disableAcc: { type: Boolean, required: true },
+});
+
+const Post = new Schema({
+  _id: { type: String, required: true }, // TODO: check correct type for objectid
```
Comment (qhanson55) [inline server/models/user-model.js:25]: we're using the Spotify username which is a string so this is all good!

Comment (me-louie) [reply server/models/user-model.js]: thanks!

File: server/models/user-model.js [hunk 2]
```diff
@@ -12,3 +12,36 @@ const Schema = mongoose.Schema;
...
+  type: { type: String, required: true },
+  time: { type: String, required: true },
```
[the remark this answers is not in the document]
Comment (me-louie) [reply server/models/user-model.js]: done!

File: server/models/user-model.js [hunk 3]
```diff
@@ -12,3 +12,38 @@ const Schema = mongoose.Schema;
...
+  disableAcc: { type: Boolean, required: true },
+});
+
+const Post = new Schema(
+  {
+    _id: { type: String, required: true }, // TODO: confirm that id type is string
+    type: { type: String, required: true },
+    content: { type: String, required: true },
+    media: { type: String, required: true }, // TODO: may need to be object for spotify?
+    usersLiked: { type: Array, required: true },
```
[the remark this answers is not in the document]
Comment (qhanson55) [reply server/models/user-model.js:30]: here are the mongodb docs:
https://mongoosejs.com/docs/populate.html

**Pull request merged**